src/functions.jl

```diff
@@ -87,9 +104,53 @@ function glyphbitmap(bmpRec::FreeType.FT_Bitmap)
     end
 
     for r = 1:bmpRec.rows
-        srcArray = pointer_to_array(row, bmpRec.width)
+        srcArray = unsafe_wrap(Array, row, bmpRec.width)
         bmp[:, r] = srcArray
         row += bmpRec.pitch
     end
     return bmp
 end
+
+"""
+    renderstring(img::AbstractArray, str::String, face, pixelsize, x0, y0;
+            halign=:hleft, valign=:vbaseline) -> Array
+
+Render `str` into `img` using the font `face` of size `pixelsize` at coordinates `x0,y0`.
+
+# Arguments
+* `halign`: :hleft, :hcenter, or :hright
+* `valign`: :vtop, :vcenter, :vbaseline, or :vbottom
+"""
+function renderstring(img::AbstractArray, str::String, face, pixelsize, x0, y0;
+            halign=:hleft, valign=:vbaseline)
+    bitmaps = Array{UInt8,2}[]
+    metrics = FontExtent{Int64}[]
+    ymin = ymax = sumadvancex = 0
+    for istr = 1:length(str)
+        bitmap, metric = renderface(face, str[istr], pixelsize)
+        push!(bitmaps, bitmap)
+        push!(metrics, metric)
+        w,h = metric.scale.values
+        bx,by = metric.horizontal_bearing
+        ymin = min(ymin,by-h)
+        ymax = max(ymax,by)
+        sumadvancex += metric.advance[1]
+    end
+
+    px = x0 - (halign==:hright ? sumadvancex : halign==:hcenter ? sumadvancex>>1 : 0)
+    py = y0 + (valign==:vtop ? ymax : valign==:vbottom ? ymin :
+            valign==:vcenter ? (ymax-ymin)>>1+ymin : 0)
+
+    for (idx,(bitmap, metric)) in enumerate(zip(bitmaps, metrics))
+        w,h = metric.scale.values
+        bx,by = metric.horizontal_bearing
+        if idx>1
+          kx, ky = kerning(str[idx-1], str[idx], face, 1.0f0)
+          px += round(Int,kx)
+        end
+        img[(1:h)+py-by, (1:w)+px-bx] = bitmap'
```

Comment:
Most likely you want `normedview(bitmap')`? Or ensure that `img` has `UInt8` eltype.

Also, this overwrites the data. Worth supporting different blending modes? Or at least thinking about what that API might look like?